### PR TITLE
Remove CRI v1alpha2

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
After the removal of dockershim we can finally also drop support for CRI v1alpha2.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
Dropped support for the Container Runtime Interface (CRI) version `v1alpha2`, which means that container runtimes just have to implement `v1`.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Looks like we have to wait until containerd 1.5.x goes EOL (still active until Oct 2022): https://github.com/containerd/containerd/blob/main/RELEASES.md

**Comment by mikebrow** [comment]
> Looks like we have to wait until containerd 1.5.x goes EOL (still active until Oct 2022): https://github.com/containerd/containerd/blob/main/RELEASES.md

or you have to require containerd v1.5.8 or above..

**Comment by saschagrunert** [comment]
Looks like we could bump the OS image there (and multiple other places): https://github.com/kubernetes/test-infra/blob/master/jobs/e2e_node/containerd/image-config.yaml#L3-L4 

Unfortunately I do not have the permissions to see which newer pipeline exists. :thinking:

**Comment by saschagrunert** [comment]
@thockin @BenTheElder do you know (or maybe can find out) which recent image familities (`pipeline-1-xx`) are available for the project `ubuntu-os-gke-cloud` ?

**Comment by bobbypage** [comment]
> @thockin @BenTheElder @bobbypage @sergeyevstifeev do you know (or maybe can find out) which recent image familities (`pipeline-1-xx`) are available for the project `ubuntu-os-gke-cloud` ?

The latest one should be `pipeline-1-24`, the latest image is `ubuntu-gke-2204-1-24-v20220623`

**Comment by rata** [comment]
@saschagrunert is there consensus on when we should move forward with this PR?

**Comment by saschagrunert** [comment]
> @saschagrunert is there consensus on when we should move forward with this PR?

Yes, when containerd 1.5 goes EOL.

**Comment by dims** [comment]
/hold cancel

(don't need hold as we need explicit approval in `cri-api`)

File: pkg/kubelet/cri/remote/remote_image.go
Comment (mikebrow) [inline pkg/kubelet/cri/remote/remote_image.go:81]: What do you think about keeping determineAPIVersion() and modifying it to report ImageFsInfo err codes.Unimplemented as not supported (and just remove the v1alpha part).  IMO there will still be a chance of old containerd's lying around that won't have v1 and we would benefit from the error log information. Additionally the info log out that we are using CRI v1 image API has some benefit. Noting we also have a possibility of image service and runtime service being split in the container runtime...

**Review by mikebrow — COMMENTED** [review]
See comment regarding possibly leaving in the v1 ImageFsInfo implemented check..

File: pkg/kubelet/cri/remote/remote_image.go
Comment (saschagrunert) [reply pkg/kubelet/cri/remote/remote_image.go]: Alright, changed the code as suggested :+1:

**Review by mikebrow — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]

**Comment by saschagrunert** [comment]
Green tests and approved PR, PTAL for lgtm @dims @mrunalp @mikebrow

**Comment by akhilerm** [comment]
I think the containerd job  (`pull-containerd-node-e2e`) to 1.5 branch is failing due to this change. containerd/containerd#7479, containerd/containerd#7633 are PRs to 1.5 branch which have been failing since last 4 days. Even after updating the machine image to `cos-stable`, the jobs are still failing.

**Pull request merged**
